### ISSUE TITLE
Should default value for cstring be nil?

### ISSUE THREAD

**Issue by StefanSalewski** [StefanSalewski/gintro#162]
```nim
proc gtk_entry_set_icon_from_icon_name(self: ptr Entry00; iconPos: EntryIconPosition;
    iconName: cstring) {.
    importc, libprag.}

proc setIconFromIconName*(self: Entry; iconPos: EntryIconPosition;
    iconName: cstring = "") =
  gtk_entry_set_icon_from_icon_name(cast[ptr Entry00](self.impl), iconPos, safeStringToCString(iconName))
```

I would guess yes.

**Comment by Dankr4d** [comment]
Makes sense. Then you can also get rid of the `safeStringToCString` call and the function itself.

**Comment by StefanSalewski** [comment]
I have done that fix and some more fixes.

You may test with

nimble uninstall gintro
nimble install gintro@#head

This is the upcomming v0.9.4 which I intend to ship this evening.

I have done a test for the gtk3 examples compiled with ARC, and for sdt compiled with ARC. Compiles and runs fine for me. Maybe I will do a few more test later this day.

**Comment by StefanSalewski** [comment]
We may have to revert this fix due to

https://github.com/nim-lang/Nim/issues/18698

**Comment by Dankr4d** [comment]
I agree.
Adding more than 1k overloaded functions is maybe not the best, but I don't know how much the compile time would increase.
Also I think we shouldn't use Options, because it bloats the code readability (and is much more to write). Options may be helpful, but I don't think that they're required in this case.
I think the "old way" is currently the best solution, because the user also doesn't need to know anything about c null string pointers.
